ts/parts-3d/Column.ts

```diff
@@ -500,27 +513,56 @@ if (seriesTypes.columnrange) {
 
 wrap(Series.prototype, 'alignDataLabel', function (
     this: Highcharts.Series,
-    proceed: Function
+    proceed: Function,
+    point: Highcharts.ColumnPoint,
+    dataLabel: Highcharts.SVGElement,
+    options: Highcharts.DataLabelsOptionsObject,
+    alignTo: Highcharts.BBoxObject
 ): void {
+    const chart = this.chart;
+
+    // In 3D we need to pass point.outsidePlot option to the justifyDataLabel
+    // method for disabling justifying dataLabels in columns outside plot
+    options.outside3dPlot = point.outside3dPlot;
 
     // Only do this for 3D columns and it's derived series
     if (
-        this.chart.is3d() &&
+        chart.is3d() &&
         this instanceof seriesTypes.column
     ) {
-        var series = this as Highcharts.ColumnSeries,
-            chart = series.chart;
-
-        var args = arguments,
-            alignTo = args[4],
-            point = args[1];
-
-        var pos = ({ x: alignTo.x, y: alignTo.y, z: series.z });
+        const series = this as Highcharts.ColumnSeries,
+            seriesOptions: Highcharts.ColumnSeriesOptions = series.options,
+            inside = pick(options.inside, !!series.options.stacking),
+            options3d = (chart.options.chart as any).options3d;
+
+        let dLPosition = {
+            x: alignTo.x + point.pointWidth / 2,
+            y: alignTo.y,
+            z: series.z + (seriesOptions as any).depth / 2
+        };
+
+        if (chart.inverted) {
+            // dLPosition.y += (point.shapeArgs as any).width / 2;
```

Comment:
Is this a comment or a commented code line?

Reply:
let's call it a ghost from a past. I will remove it in the next commit.